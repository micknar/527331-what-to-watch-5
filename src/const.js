export const filmsCount = {
  SIMILAR: 4,
  PER_STEP: 8,
};

export const RatingMarkType = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const FilmPageNav = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  FILMS: `/films/`,
  FILMS_ID: `/films/:id`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  REVIEW: `/films/:id/review`,
  ROOT: `/`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id`,
};

export const APIRoute = {
  COMMENTS: `/comments/`,
  FILMS: `/films/`,
  LOGIN: `/login`,
  PROMO_FILM: `/films/promo`,
};
