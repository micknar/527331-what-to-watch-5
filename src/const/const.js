export const HOUR = 60;
export const WHITE_RGB_COLOR_CODE = 255;
export const CARD_HOVER_TIMEOUT = 1000;

export const RATING_MARKS = [1, 2, 3, 4, 5];

export const ratingMarkMap = {
  'bad': {
    text: `Bad`,
    value: 3,
  },
  'normal': {
    text: `Normal`,
    value: 5,
  },
  'good': {
    text: `Good`,
    value: 8,
  },
  'veryGood': {
    text: `Very good`,
    value: 10,
  },
  'awesome': {
    text: `Awesome`,
    value: 10,
  },
};

export const genreMap = {
  'comedy': {
    single: `Comedy`,
    plural: `Comedies`,
  },
  'drama': {
    single: `Drama`,
    plural: `Dramas`,
  },
  'thriller': {
    single: `Thriller`,
    plural: `Thrillers`,
  },
};

export const FilmsCount = {
  SIMILAR: 4,
  PER_STEP: 8,
};

export const FilmPageNav = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  FILMS: `/films/`,
  FILMS_ID: `/films/:id`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  REVIEW: `/films/:id/review`,
  ROOT: `/`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id`,
};

export const APIRoute = {
  COMMENTS: `/comments/`,
  FAVORITE_FILMS: `/favorite`,
  FILMS: `/films/`,
  LOGIN: `/login`,
  PROMO_FILM: `/films/promo`,
};

export const FavoriteStatusCode = {
  ADD: 1,
  REMOVE: 0,
};

export const NameSpace = {
  APP_STATE: `APP_STATE`,
  USER: `USER`,
};

export const Radix = {
  DECIMAL: 10,
  HEXADECIMAL: 16,
};

export const HttpCode = {
  OK: 200,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
};
