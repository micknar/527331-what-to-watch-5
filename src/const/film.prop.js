import PropTypes from "prop-types";

export default PropTypes.shape({
  id: PropTypes.number,
  name: PropTypes.string,
  posterImage: PropTypes.string,
  previewImage: PropTypes.string,
  backgroundImage: PropTypes.string,
  backgroundColor: PropTypes.string,
  videoLink: PropTypes.string,
  previewVideoLink: PropTypes.string,
  description: PropTypes.string,
  rating: PropTypes.number,
  scoresCount: PropTypes.number,
  ratingMark: PropTypes.string,
  director: PropTypes.string,
  starring: PropTypes.arrayOf(PropTypes.string),
  runTime: PropTypes.shape({
    hours: PropTypes.number,
    minutes: PropTypes.number,
  }),
  genre: PropTypes.string,
  released: PropTypes.number,
  isFavorite: PropTypes.bool,
});
