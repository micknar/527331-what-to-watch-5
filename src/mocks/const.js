export const FILMS_COUNT = 17;
export const FILMS_COUNT_PER_STEP = 8;
export const RATING_MIN = 0;
export const RATING_MAX = 10;
export const DIRECTORS = [`Anthony Mann`, `Martin Scorsese`, `Clint Eastwood`, `Ilya Naishuller`, `Guy Ritchie`];
export const ACTORS = [`Erich von Stroheim`, `Mary Beth Hughes`, `Dan Duryea`, `Robert Duvall`, `Liam Neeson`, `Russell Crowe`, `Woody Harrelson`, `Bill Murray`, `Edward Norton`, `Willem Dafoe`, `Tilda Swinton`];
export const USERS = [`John Doe`, `Tim Macoveev`, `Bill Murray`, `Jude Law`, `Tony Revoloru`, `Owen Wilkinson`, `Jeff Goldblum`];

export const GENRES = [
  `Comedy`,
  `Crime`,
  `Documentary`,
  `Drama`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thriller`,
];

export const TITLES = [
  `Aviator`,
  `Bohemian Rhapsody`,
  `Dardjeeling Limited`,
  `Fantastic Beasts The Crimes Of Grindelwald`,
  `Johnny English`,
  `Macbeth`,
  `Midnight Special`,
  `Mindhunter`,
];

export const POSTERS = [
  `/img/aviator.jpg`,
  `/img/bohemian-rhapsody.jpg`,
  `/img/dardjeeling-limited.jpg`,
  `/img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `/img/johnny-english.jpg`,
  `/img/macbeth.jpg`,
  `/img/midnight-special.jpg`,
  `/img/mindhunter.jpg`,
];

export const TEXTS = [
  `Lorem ipsum dolor sit amet, consectetur adipiscing elit.`,
  `Cras aliquet varius magna, non porta ligula feugiat eget.`,
  `Fusce tristique felis at fermentum pharetra.`,
  `Aliquam id orci ut lectus varius viverra.`,
  `Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.`,
  `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
  `Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
  `Sed sed nisi sed augue convallis suscipit in sed felis.`,
  `Aliquam erat volutpat.`,
  `Nunc fermentum tortor ac porta dapibus.`,
  `In rutrum ac purus sit amet tempus.`,
];
