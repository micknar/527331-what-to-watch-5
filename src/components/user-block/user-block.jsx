import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute, AuthorizationStatus} from "../../const";

const UserBlock = (props) => {
  const {authorizationStatus} = props;

  return (
    <div className="user-block">
      {
        authorizationStatus === AuthorizationStatus.AUTH
          ? <div className="user-block__avatar">
            <Link to={AppRoute.MY_LIST}>
              <img src="/img/avatar.jpg" alt="User avatar" width="63" height="63" />
            </Link>
          </div>
          : <Link to={AppRoute.LOGIN} className="user-block__link">Sign in</Link>
      }
    </div>
  );
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus
});

UserBlock.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
};

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
