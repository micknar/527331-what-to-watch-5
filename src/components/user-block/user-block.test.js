import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {UserBlock} from "./user-block";
import {AuthorizationStatus} from "../../const/const";

const avatar = `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`;

describe(`Should UserBlock render correctly`, () => {
  it(`With guest`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              avatar={avatar}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With authorized user`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              authorizationStatus={AuthorizationStatus.AUTH}
              avatar={avatar}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
