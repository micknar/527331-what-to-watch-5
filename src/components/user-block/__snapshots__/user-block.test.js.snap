// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly With authorized user 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg"
        width="63"
      />
    </a>
  </div>
</div>
`;

exports[`Should UserBlock render correctly With guest 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
